### PR TITLE
Test source before geting the key [preventing Object.keys(false)]

### PR DESCRIPTION
**Pull request by namiltd**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Object.keys(false) fix
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR         | No
| Any Dependency Changes?  | No
| License                  | MIT

### CODE REVIEW

**Comment by loganfsmyth** [comment]
A lot of the time Babel assumes that you've loaded a polyfill. For instance core-js polyfills `Object.keys` to handle things like this.

**Comment by namiltd** [comment]
Optimized

File: packages/babel-helpers/src/helpers.js [hunk 1]
```diff
@@ -377,7 +377,7 @@ helpers.objectSpread = () => template.program.ast`
 
   export default function _objectSpread(target) {
     for (var i = 1; i < arguments.length; i++) {
-      var source = (arguments[i] != null) ? arguments[i] : {};
+      var source = arguments[i] != null && arguments[i] !== false ? arguments[i] : {};
```
Comment (jridgewell) [inline packages/babel-helpers/src/helpers.js]: We can fix this and speed up all non-objets:

```js
var source = arguments[i];
if (source !== Object(source)) continue;

var ownKeys = Object.keys(source);
//...

Comment (nicolo-ribaudo) [reply packages/babel-helpers/src/helpers.js]: I believe that strings should be spread.
The check could be `if (!source) continue;`

Comment (loganfsmyth) [reply packages/babel-helpers/src/helpers.js]: If we do want to do this, which I'm not convinced we do, it seems like we should approach it as including our own polyfill for `Object.keys`, but really it seems not too hard to tell people to load their own polyfills?

**Comment by namiltd** [comment]
Implemented filtering of non-objects.

**Comment by namiltd** [comment]
I have tested, the 'source !== Object (source)' works correctly

File: packages/babel-helpers/src/helpers.js (hunk 1 above)
Comment (namiltd) [reply packages/babel-helpers/src/helpers.js]: I discovered the problem in the Bootstrap 4 framework that uses babel.

**Comment by namiltd** [comment]
I understand, therefore, that this code has little chance of being introduced, which is a pity.

**Comment by namiltd** [comment]
The problem has disappeared, probably https://github.com/babel/babel/pull/7777 blocking this type of calls.

**Pull request closed without merging**